azext_iot/digitaltwins/providers/resource.py

```diff
@@ -495,3 +496,139 @@ def delete_private_endpoint_conn(self, name, conn_name, resource_group_name=None
             )
         except ErrorResponseException as e:
             raise CLIError(unpack_msrest_error(e))
+
+    def create_adx_data_connection(
+        self,
+        name,
+        conn_name,
+        adx_cluster_name,
+        adx_database_name,
+        eh_namespace,
+        eh_entity_path,
+        adx_table_name=None,
+        adx_resource_group=None,
+        adx_subscription=None,
+        eh_consumer_group="$Default",
+        eh_resource_group=None,
+        eh_subscription=None,
+        resource_group_name=None,
+        yes=False,
+    ):
+        target_instance = self.find_instance(
+            name=name, resource_group_name=resource_group_name
+        )
+        if not resource_group_name:
+            resource_group_name = self.get_rg(target_instance)
+        subscription = target_instance.id.split("/")[2]
+
+        if len(conn_name) <= 2:
+            raise CLIError(
+                "The connection name must have a length greater than 2"
+            )
+
+        adx_resource_group = adx_resource_group if adx_resource_group else resource_group_name
+        eh_resource_group = eh_resource_group if eh_resource_group else resource_group_name
+        adx_subscription = adx_subscription if adx_subscription else subscription
+        eh_subscription = eh_subscription if eh_subscription else subscription
+
+        from azext_iot.digitaltwins.providers.connection.builders import build_adx_connection_properties
+        properties = build_adx_connection_properties(
+            adx_cluster_name=adx_cluster_name,
+            adx_database_name=adx_database_name,
+            adx_table_name=adx_table_name,
+            adx_resource_group=adx_resource_group,
+            adx_subscription=adx_subscription,
+            dt_instance=target_instance,
+            eh_namespace=eh_namespace,
+            eh_entity_path=eh_entity_path,
+            eh_consumer_group=eh_consumer_group,
+            eh_resource_group=eh_resource_group,
+            eh_subscription=eh_subscription,
+            yes=yes,
+        )
+
+        try:
+            def check_state(lro):
```

Comment:
Revisit of DT RP issues with msrest? We should combine as much of the duplicate code as possible from our workaround in check_state used in the instance creation method.

Reply:
Will write a generalized ms_rest check_state - depending on the resource, the provisioning state can be in the main level or nested in properties.

Reply:
AFAIK, thus far we've only run into polling issues with ADT resources so it can be scoped to DT.

Reply:
There are two additional issues:
the error message will have to be more generic (something like "Please monitor the status of the created instance with the show command.")
the amount of polling for the dt instance creation vs the data history creation is very different (5 times for the DT create vs 20 times for data history)